components/TaskCard/TaskCard.tsx

```diff
@@ -22,14 +22,23 @@ const frequencyList: Record<FrequencyType | string, string> = {
  * @param {object} props.taskData - 할 일 데이터
  * @returns {JSX.Element} 할 일 카드 컴포넌트
  */
-function TaskCard({ type, taskData, updateTask }: TaskCardProps) {
+function TaskCard({ type, taskData, updateTask, onClick }: TaskCardProps) {
   const { id, name, description, date, doneAt, commentCount, frequency } =
     taskData;
   const [isChecked, setIsChecked] = useState(Boolean(doneAt));
   const isTaskList = type === 'taskList';
   const frequencyText = frequencyList[frequency];
+  const checkboxRef = useRef<HTMLInputElement>(null);
 
-  const handleCheckedToggle = () => {
+  const handleClickTaskCard = (event: MouseEvent<HTMLDivElement>) => {
+    if (checkboxRef.current?.contains(event.target as Node)) {
+      if (onClick) onClick(null);
+      return;
+    }
+    if (onClick) onClick(taskData);
+  };
```

Comment:
여기도 명시적으로 타입 지정을 해주는것이 좋을것같습니다
```suggestion
    if (onClick) onClick(taskData as ITask);
```